### PR TITLE
fix version check with hash version

### PR DESCRIPTION
**Pull request by 3pointer**
### What problem does this PR solve? 
Fix version check when version has hash suffix.

### What is changed and how it works?
remove version hash to compare version

### Check List 

Tests 

 - Unit test

Related changes

 - Need to cherry-pick to the release branch

### Release Note

 - No release note

### CODE REVIEW

File: pkg/utils/version.go [hunk 1]
```diff
@@ -53,13 +54,18 @@ var minTiKVVersion *semver.Version = semver.New("3.1.0-beta.2")
 var incompatibleTiKVMajor3 *semver.Version = semver.New("3.1.0")
 var incompatibleTiKVMajor4 *semver.Version = semver.New("4.0.0-rc.1")
 
-func removeV(v string) string {
+func removeVAndHash(v string) string {
+	hash := regexp.MustCompile("-[0-9]+-g[0-9a-f]{7,}")
```
Comment (kennytm) [inline pkg/utils/version.go:58]: move this outside of the function (make it a global)

File: pkg/utils/version.go [hunk 2]
```diff
@@ -53,13 +54,18 @@ var minTiKVVersion *semver.Version = semver.New("3.1.0-beta.2")
...
+	if hash.Match([]byte(v)) {
+		v = hash.ReplaceAllLiteralString(v, "")
+	}
```
Comment (kennytm) [inline pkg/utils/version.go:61]:
```suggestion
	v = hash.ReplaceAllLiteralString(v, "")
```

File: pkg/restore/client.go [hunk 3]
```diff
@@ -504,6 +504,7 @@ func (rc *Client) RestoreFiles(
 			return err
 		}
 	}
+	wg.Wait()
```
Comment (overvenus) [inline pkg/restore/client.go:507]: This is a bugfix?

Comment (3pointer) [reply pkg/restore/client.go:507]: yes, I think it fix the updateCh send on close channel at [L490](https://github.com/pingcap/br/blob/1cda453e3feb265b223eb032225254b1ff3d5ab8/pkg/restore/client.go#L490), when last file import success, if will wait until updateCh.Inc finish.

**Pull request merged**